velox/dwio/common/CMakeLists.txt

```diff
@@ -22,25 +22,37 @@ elseif(${VELOX_BUILD_TEST_UTILS})
 endif()
 
 add_library(
-  velox_dwio_common
-  BitConcatenation.cpp
-  BitPackDecoder.cpp
+  velox_dwio_common_base
   BufferedInput.cpp
   CachedBufferedInput.cpp
   CacheInputStream.cpp
-  ColumnSelector.cpp
   DataBufferHolder.cpp
+  FileSink.cpp
+  InputStream.cpp
+  IoStatistics.cpp
+  OutputStream.cpp
+  SeekableInputStream.cpp)
+target_link_libraries(
+  velox_dwio_common_base
+  velox_dwio_common_exception
+  velox_buffer
+  velox_caching
+  velox_exception
+  velox_memory
+  Folly::folly)
+
+add_library(
+  velox_dwio_common
```

Comment:
@assignUser I believe this is the minimal split that needs to be done to separate out the cyclical dependency. Now this causes a new lib to be created. From the naming convention perspective it should be in its own subdir with a perhaps more fitting name to what is included (maybe velox_dwio_common_stream_buffer).
But any dir change causes code changes (due to the include files used in the code).

I suppose we would need to discuss with Pedro or someone else how to approach it?

Reply:
Thanks for working on this! Yes @pedroerp and @kgpai would be the right people to discuss this.

Reply:
@kgpai and @pedroerp Any thoughts on how to approach this from a code organization perspective to resolve the cycling dependencies? I believe a new library with a subset needs to be created that is then used (as proposed in this PR). But it needs to be organized accordingly. I'm looking for some pointers because reorganizing is big if directory changes are involved.

Reply:
Unless I'm missing something terribly obvious, we don't really need the velox_dwio_common -> velox_dwio_common_compression dependency. 

https://github.com/facebookincubator/velox/pull/6760 works on my laptop, at least.

Reply:
Right, I assumed the intent was to have this part of `velox_dwio_common` just like it was done with the other libraries `velox_dwio_common_*` to have everything in a larger library. Keeping it separate like you have done works too.